### PR TITLE
Bump minimum PHP support to 7.4

### LINKED ISSUE #791: Minimum PHP version bump
Current: **7.2**

Sparked from https://github.com/WordPress/plugin-check/pull/789

I am working on updating the readme parser to use its latest version. That class is coming from WP.org and its minimum PHP support is `7.4`. So we are now stuck the older version of the readme parser in our plugin.

I created ticket in meta about the issue but it did not get traction to make parser class to make PHP 7.2 compatible - https://meta.trac.wordpress.org/ticket/7749

So, may be we could bump our min PHP version instead of using composer patch for the parser.

I have not seen the discussion about minimum PHP version support in PCP particularly. What philosophy we are following for the minimum PHP version support?

### PR DESCRIPTION
**Pull request by ernilambar**
Fixes https://github.com/WordPress/plugin-check/issues/791

### CODE REVIEW

**Review by davidperezgar — APPROVED** [review]

**Comment by swissspidy** [comment]
Can we do PHP 8.0+ instead? Would be more reasonable for a developer tool like this

**Comment by ernilambar** [comment]
> Can we do PHP 8.0+ instead? Would be more reasonable for a developer tool like this

Directory run in 7.4 I guess. It wont be feasible to run PCP in the directory if we bump to 8.0 now.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ Thank you for your interest in contributing to the Plugin Check plugin!
 In general, all code must follow the [WordPress Coding Standards and best practices](https://developer.wordpress.org/coding-standards/). All code in the Plugin Check plugin must follow these requirements:
 
 - **WordPress**: The plugin's minimum WordPress version requirement is 6.3.
-- **PHP**: Always match the latest WordPress version. The minimum required version right now is 7.2.24.
+- **PHP**: The plugin's minimum PHP version requirement is 7.4.
```
Comment (mukeshpanchal27) [inline CONTRIBUTING.md]:
```suggestion
- **PHP**: The minimum required version right now is 7.4.
```

**Comment by mukeshpanchal27** [comment]
Per https://wordpress.org/about/stats/, After these PR get merged the total `~11.14%` sites are not able to use PCP.

<table class="google-visualization-table-table" cellspacing="0" style="height: 450px;"><thead><tr class="google-visualization-table-tr-head"><th class="google-visualization-table-th gradient unsorted">Version<span class="google-visualization-table-sortind"></span></th><th class="google-visualization-table-th gradient unsorted">Usage<span class="google-visualization-table-sortind"></span></th></tr></thead><tbody><tr class="google-visualization-table-tr-even"><td class="google-visualization-table-td">7.3</td><td class="google-visualization-table-td google-visualization-table-td-number">3.30%</td></tr><tr class="google-visualization-table-tr-odd"><td class="google-visualization-table-td">7.2</td><td class="google-visualization-table-td google-visualization-table-td-number">2.78%</td></tr><tr class="google-visualization-table-tr-even"><td class="google-visualization-table-td">7.1</td><td class="google-visualization-table-td google-visualization-table-td-number">0.70%</td></tr><tr class="google-visualization-table-tr-odd"><td class="google-visualization-table-td">7.0</td><td class="google-visualization-table-td google-visualization-table-td-number">1.12%</td></tr><tr class="google-visualization-table-tr-even"><td class="google-visualization-table-td">5.6</td><td class="google-visualization-table-td google-visualization-table-td-number">2.28%</td></tr><tr class="google-visualization-table-tr-odd"><td class="google-visualization-table-td">5.5</td><td class="google-visualization-table-td google-visualization-table-td-number">0.26%</td></tr><tr class="google-visualization-table-tr-even"><td class="google-visualization-table-td">5.4</td><td class="google-visualization-table-td google-visualization-table-td-number">0.31%</td></tr><tr class="google-visualization-table-tr-odd"><td class="google-visualization-table-td">5.3</td><td class="google-visualization-table-td google-visualization-table-td-number">0.39%</td></tr><tr class="google-visualization-table-tr-even"><td class="google-visualization-table-td">5.2</td><td class="google-visualization-table-td google-visualization-table-td-number">0.17%</td></tr></tbody></table>

**Comment by ernilambar** [comment]
Since this is developer targeted plugin, we may not have to stick with core's PHP philosophy, I guess.

**Review by mukeshpanchal27 — APPROVED** [review]

**Pull request merged**